src/components/qmlexecutor/gui/qmlexecutorguiimpl.h

```diff
@@ -96,27 +97,36 @@ public slots:
     {
         if(url.isValid() && !url.isEmpty())
         {
-            _qmlUrl = url;
+            // The easiest way to reload QML seems to be creating new QQuickWidget
```

Comment:
loadQML function is aware of that so no fuss is needed here

Reply:
Not an expert here but I was getting a lot of warnings about NULL reference when reloading QML (although new QML was working properly...). It seems to me that clearComponentCache that was used to clear engine was not clearing context properties.